util/mkbuildinf.pl

```diff
@@ -1,26 +1,43 @@
 #! /usr/bin/env perl
-# Copyright 2014-2016 The OpenSSL Project Authors. All Rights Reserved.
+# Copyright 2014-2017 The OpenSSL Project Authors. All Rights Reserved.
 #
 # Licensed under the OpenSSL license (the "License").  You may not use
 # this file except in compliance with the License.  You can obtain a copy
 # in the file LICENSE in the source distribution or at
 # https://www.openssl.org/source/license.html
 
+use strict;
+use warnings;
 
 my ($cflags, $platform) = @ARGV;
-
 $cflags = "compiler: $cflags";
-$date = localtime();
+
+my $date = $ENV{'SOURCE_DATE_EPOCH'} || localtime();
```

Comment:
`$ENV{'SOURCE_DATE_EPOCH'}` is the epoch, i.e. just a simple integer, which should be fed to `localtime`, so this is what that assignment should look like:

``` perl
$date = localtime($ENV{'SOURCE_DATE_EPOCH'} // time());
```

Also, we might want to consider `gmtime` over `localtime` and make sure we add `" UTC"` to the date.

Reply:
fixed, and converted to gmtime; thanks.  additional commit pushed.